File: backend/src/HowAreMyFinances.Api/Models/Month.cs
namespace HowAreMyFinances.Api.Models;

public sealed record Month(
    Guid Id,
    Guid UserId,
    int Year,
    int MonthNumber,
    decimal Salary,
    string? Notes,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record CategoryBreakdownItem(
    Guid CategoryId,
    string CategoryName,
    decimal Total
);

public sealed record MonthDetail(
    Guid Id,
    Guid UserId,
    int Year,
    int MonthNumber,
    decimal Salary,
    string? Notes,
    decimal TotalSpent,
    decimal PlannedSpent,
    decimal TotalIncome,
    decimal Remaining,
    IReadOnlyList<CategoryBreakdownItem> CategoryBreakdown,
    int DaysRemaining,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record MonthSummary(
    Guid Id,
    Guid UserId,
    int Year,
    int MonthNumber,
    decimal Salary,
    string? Notes,
    decimal TotalSpent,
    decimal TotalIncome,
    decimal Remaining,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record CreateMonthRequest(int Year, int Month, decimal Salary);

public sealed record UpdateMonthRequest(decimal? Salary, string? Notes);

File: shared/types/month.ts
export interface Month {
  id: string;
  userId: string;
  year: number;
  monthNumber: number;
  salary: number;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface MonthSummary extends Month {
  totalSpent: number;
  totalIncome: number;
  remaining: number;
}

export interface CategoryBreakdownItem {
  categoryId: string;
  categoryName: string;
  total: number;
}

export interface MonthDetail extends Month {
  totalSpent: number;
  plannedSpent: number;
  totalIncome: number;
  remaining: number;
  categoryBreakdown: CategoryBreakdownItem[];
  daysRemaining: number;
}

export interface CreateMonthRequest {
  year: number;
  month: number;
  salary: number;
}

export interface UpdateMonthRequest {
  salary?: number;
  notes?: string;
}
